File: kotlin/lib/src/main/kotlin/models/ListResponseMessageOut.kt
// This file is @generated
package com.svix.kotlin.models

import kotlinx.serialization.Serializable

@Serializable
data class ListResponseMessageOut(
    val data: List<MessageOut>,
    val done: Boolean,
    val iterator: String? = null,
    val prevIterator: String? = null,
)

File: javascript/src/models/listResponseMessageOut.ts
// this file is @generated
import { type MessageOut, MessageOutSerializer } from "./messageOut";

export interface ListResponseMessageOut {
  data: MessageOut[];
  iterator: string | null;
  prevIterator?: string | null;
  done: boolean;
}

export const ListResponseMessageOutSerializer = {
  _fromJsonObject(object: any): ListResponseMessageOut {
    return {
      data: object["data"].map((item: MessageOut) =>
        MessageOutSerializer._fromJsonObject(item)
      ),
      iterator: object["iterator"],
      prevIterator: object["prevIterator"],
      done: object["done"],
    };
  },

  _toJsonObject(self: ListResponseMessageOut): any {
    return {
      data: self.data.map((item) => MessageOutSerializer._toJsonObject(item)),
      iterator: self.iterator,
      prevIterator: self.prevIterator,
      done: self.done,
    };
  },
};
